### DISCUSSION TITLE
Command Line Attributes Not Parsed Correctly by Command Line Tools in Windows

### DISCUSSION THREAD

**Discussion by ep5000** [dcm4che/dcm4che#1281] [Q&A] [answered]
When using the command line tools, settings parameter attributes is ok in Linux but in Windows the attributes are dropped.  See the following example:

```
In Linux (bash):

# This works
storescu -c myaet@localhost:4242 -s PatientName=Mike myfile.dcm
```

```
In Windows (PowerShell or CMD):

# This does not work
storescu -c myaet@localhost:4242 -s PatientName=Mike myfile.dcm
```

In Windows the following error is received:

<pre>
java.lang.IllegalArgumentException: Mike
        at org.dcm4che3.tool.common.CLIUtils.toTag(CLIUtils.java:686)
        at org.dcm4che3.tool.common.CLIUtils.toTags(CLIUtils.java:676)
        at org.dcm4che3.tool.common.CLIUtils.addAttributes(CLIUtils.java:804)
</pre>

It appears that since the command line tools are batch files the '=' between the parameter key and value is dropped thus causing the issue.

**Comment by gunterze** [comment]
Guess you need to quote the arguments containing `=`

**Comment by ep5000** [comment]
I tried the following and none worked:

`storescu -c myaet@localhost:4242 -s 'PatientName=Mike' myfile.dcm
`
`storescu -c myaet@localhost:4242 -s "PatientName=Mike" myfile.dcm
`
`storescu -c myaet@localhost:4242 '-sPatientName=Mike' myfile.dcm
`

They all throw the same exception:

<pre>
java.lang.IllegalArgumentException: Mike
        at org.dcm4che3.tool.common.CLIUtils.toTag(CLIUtils.java:686)
        at org.dcm4che3.tool.common.CLIUtils.toTags(CLIUtils.java:676)
        at org.dcm4che3.tool.common.CLIUtils.addAttributes(CLIUtils.java:804)
</pre>

**Comment by gunterze** [comment, marked as answer]
You have to separate options which support multiple values from the program  arguments by `--`.

```
storescu -c myaet@localhost:4242 -s 'PatientName=Mike' -- myfile.dcm
```

**Comment by ep5000** [comment]
Ok that works thanks.  Just as a note, this seems to work within CMD on Windows only (not PowerShell).  Also for me using double quotes is working (not single quotes).  For example:

The following doesn't work (via CMD on Windows nor PowerShell):

`storescu -c myaet@localhost:4242 -s 'PatientName=Mike' -- myfile.dcm`

The above produces the following error:
<pre>
java.lang.IllegalArgumentException: 'PatientName
        at org.dcm4che3.tool.common.CLIUtils.toTag(CLIUtils.java:686)
        at org.dcm4che3.tool.common.CLIUtils.toTags(CLIUtils.java:676)
        at org.dcm4che3.tool.common.CLIUtils.addAttributes(CLIUtils.java:804)
        at org.dcm4che3.tool.storescu.StoreSCU.main(StoreSCU.java:257)
</pre>

The following **works** on Windows via CMD on Windows (not PowerShell):

`storescu -c myaet@localhost:4242 -s "PatientName=Mike" -- myfile.dcm`


Thanks for the info: I'll stick to CMD and double-quotes.  cheers
